multicluster/controllers/multicluster/stale_controller.go

```diff
@@ -263,6 +296,44 @@ func (c *StaleResCleanupController) cleanupServiceResourceExport(commonArea comm
 		delete(staleResExpItems, se.Namespace+"/"+se.Name+"endpoint")
 	}
 
+	for _, r := range staleResExpItems {
+		re := r
+		klog.InfoS("Cleaning up stale ResourceExport", "ResourceExport", klog.KObj(&re))
+		if err := commonArea.Delete(ctx, &re, &client.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *StaleResCleanupController) cleanLabelIdentityResourceExport(commonArea commonarea.RemoteCommonArea,
+	resExpList *mcsv1alpha1.ResourceExportList) error {
+	podList, nsList := &corev1.PodList{}, &corev1.NamespaceList{}
+	if err := c.List(ctx, podList, &client.ListOptions{}); err != nil {
+		return err
+	}
+	if err := c.List(ctx, nsList, &client.ListOptions{}); err != nil {
+		return err
+	}
+	allResExpItems := resExpList.Items
+	staleResExpItems := map[string]mcsv1alpha1.ResourceExport{}
+	for _, resExp := range allResExpItems {
+		if resExp.Spec.Kind == common.LabelIdentityKind && resExp.Labels[common.SourceClusterID] == c.localClusterID {
+			staleResExpItems[resExp.Spec.LabelIdentity.NormalizedLabel] = resExp
+		}
+	}
+	nsLabels := map[string]string{}
+	for _, ns := range nsList.Items {
```

Comment:
You added `nsLabels[v1.LabelMetadataName] = ns` for K8s earlier than v1.21, I think you also need to add it before handle deletion?